components/gitpod-db/src/typeorm/user-db-impl.ts

```diff
@@ -640,7 +652,11 @@ export class TypeORMUserDBImpl implements UserDB {
 }
 
 export class TransactionalUserDBImpl extends TypeORMUserDBImpl {
```

Comment:
Do we really need the inheritance here?

What prevents us from making this part of the regular `TypeORMUserDBImpl`?

Reply:
Because everything should run in the same transaction the state (connection) needs to be passed. This is what that class does. I didn't invent this in this PR, though. So would love to keep it out of the discussion